.circleci/config.yml

```diff
@@ -107,27 +114,48 @@ jobs:
       - run: yarn build:webpack
 
   build-e2e:
+    working_directory: ~/vets-website
     docker:
       - image: cypress/base:10.15.3
     steps:
       - checkout
       - restore_cache:
           keys:
-            - yarn-packages-{{ checksum "yarn.lock" }}
+            - yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}
+      - restore_cache:
+          keys:
+            - node-v1-{{ .Branch }}-{{ checksum "package.json" }}
       - run:
-          command: yarn install --frozen-lockfile
+          name: Install Dependencies
+          command: yarn --frozen-lockfile --cache-folder ~/.cache/yarn
+      - run:
+          name: Verify Cypress is Installed
+          command: yarn cypress verify || yarn cypress install
       - save_cache:
-          key: yarn-packages-{{ checksum "yarn.lock" }}
+          key: yarn-packages-{{ .Branch }}-{{ checksum "yarn.lock" }}
```

Comment:
Why did this change to be branch-specific? I thought the lock file would be sufficient for keeping the right dependencies cached.

Reply:
Ideally `checksum` should be sufficient for caching the right dependencies. I'm trying to come up with examples where using only `checksum` would fall short, but I can't think of any 🤔. I noticed it was a common pattern to add branch specific caching. From examples, it seemed like best practice to add that extra layer, but it might not be necessary.

Reply:
I removed the branch name from the cache key. The lock file should be sufficient enough for now.